### PR TITLE
Add one new action to establish baremetal server eCPI environment

### PR DESCRIPTION
**Pull request by xingzhou**
In this patch, we added the following features:

1. Add a new action named "establish_bare_metal_env" to set up a eCPI env on a newly created bare_metal server.

2. Finished the first step of "establish_bare_metal_env" action which is to use softlayer-go's hardware service to create a new baremetal server.

Signed-off-by: xingzhou <xingzhou@cn.ibm.com>

### CODE REVIEW

File: dev/establish_bare_metal_env.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+	"method": "establish_bare_metal_env",
+	"arguments": [],
+	"context": {
+		"director_uuid": "3f695519-5a17-480f-879a-582dbe31131e"
```
Comment (maximilien) [inline dev/establish_bare_metal_env.json]: Why is director UUID hardcoded here?

File: softlayer/baremetal/baremetal_creator.go [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+		HourlyBillingFlag:            true,
+		OperatingSystemReferenceCode: ostype,
+
+		Datacenter: &datatypes.Datacenter{
+			Name: datacenter,
+		},
+	}
+
+	baremetal, err := service.CreateObject(template)
+
```
Comment (maximilien) [inline softlayer/baremetal/baremetal_creator.go]: no need for new line here. Consider if err check part of previous statement :)

File: softlayer/baremetal/baremetal_creator_test.go [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	boshlog "github.com/cloudfoundry/bosh-agent/logger"
+	common "github.com/maximilien/bosh-softlayer-cpi/common"
+	bm "github.com/maximilien/bosh-softlayer-cpi/softlayer/baremetal"
+	fakeslclient "github.com/maximilien/softlayer-go/client/fakes"
+)
+
+var _ = Describe("BaremetalCreator", func() {
```
Comment (maximilien) [inline softlayer/baremetal/baremetal_creator_test.go]: Can we have more tests for the BM creator? Seems like at least one test to verify bad parameters would be needed?

File: softlayer/baremetal/baremetal_finder_test.go [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+		logger = boshlog.NewLogger(boshlog.LevelNone)
+
+		finder = bm.NewBaremetalFinder(
+			softLayerClient,
+			logger,
+		)
+		common.SetTestFixtureForFakeSoftLayerClient(softLayerClient, "SoftLayer_Hardware_Service_getObject.json")
+	})
+
+	Describe("Find", func() {
```
Comment (maximilien) [inline softlayer/baremetal/baremetal_finder_test.go]: Same here we should have tests for negative paths, e.g., not able to find BM servers and making sure error is returned and propagated.

File: softlayer/baremetal/baremetal_finder_test.go [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+		Context("succeeded", func() {
```
Comment (maximilien) [inline softlayer/baremetal/baremetal_finder_test.go]: Also, for successful case you should have expectations for the different attributes for the created and found BM server

**Comment by maximilien** [comment]
Can you address the issues I listed. Mostly adding additional tests. Should be quick. Then resubmit PR and I will accept and merge it. Thanks.

**Comment by xingzhou** [comment]
Max, please take a look at this new PR based on your comments. Thanks!

File: action/establish_bare_metal_env.go [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+const (
+	MEMORY_CON     = "SL_BARE_METAL_MEMORY"
+	PROCESSOR_CON  = "SL_BARE_METAL_PROCESSOR"
+	DISK_CON       = "SL_BARE_METAL_DISK"
+	HOST_CON       = "SL_BARE_METAL_HOST"
+	DOMAIN_CON     = "SL_BARE_METAL_DOMAIN"
+	OS_CON         = "SL_BARE_METAL_OS"
+	DATACENTER_CON = "SL_DATA_CENTER"
+
+	MAX_RETRIES = 120
```
Comment (maximilien) [inline action/establish_bare_metal_env.go]: why 120? is there a reason? just curious

File: action/establish_bare_metal_env.go [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+func NewEstablishBareMetalEnv(bmCreator bm.BaremetalCreator, bmFinder bm.BaremetalFinder) EstablishBareMetalEnv {
+	return EstablishBareMetalEnv{
+		bmCreator: bmCreator,
+		bmFinder:  bmFinder,
+	}
+}
+
+func (b EstablishBareMetalEnv) Run() (interface{}, error) {
+
+	// Step #1, use baremetal creator to create a new baremetal server
```
Comment (maximilien) [inline action/establish_bare_metal_env.go]: I wonder why are these set as environment variables vs `VMProperties`?

**Comment by maximilien** [comment]
OK cool. Just one comment. We can always change this if it makes sense.

**Pull request merged**